server/.eslintrc.js

```diff
@@ -25,6 +25,17 @@ module.exports = {
     'unicorn/prefer-top-level-await': 'off',
     'unicorn/prefer-event-target': 'off',
     'unicorn/no-thenable': 'off',
+    '@typescript-eslint/await-thenable': 'error',
+    '@typescript-eslint/no-floating-promises': 'error',
+    '@typescript-eslint/no-misused-promises': [
+      'error',
+      {
+        checksVoidReturn: false,
```

Comment:
we should probably try to avoid disabling this. You might find something like `handlePromiseError` from the web project's `$lib/utils` to be useful

Reply:
I wasn't able to get it working with that removed, but I did check with it enabled you'd like to take a stab at this. I'm going to bed for now

Reply:
I pushed a commit for it

Reply:
Thanks, but there's a strange error when running in github actions. Locally, everything compiles but not when the tests are running in github.